ecosystem/typescript/sdk/src/plugins/aptos_token.ts

```diff
@@ -0,0 +1,490 @@
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "remove_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), propertyKey],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Update token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key stored on-chain
+   * @param type the property typed stored on-chain
+   * @param value the property value to be stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async updateTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    propertyKey: string,
+    propertyType: PropertyType,
+    propertyValue: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "update_property",
+      [tokenType || this.tokenType],
+      [
+        HexString.ensure(token).hex(),
+        propertyKey,
+        PropertyTypeMap[propertyType],
+        getSinglePropertyValueRaw(propertyValue, PropertyTypeMap[propertyType]),
+      ],
+      extraArgs,
+    );
+  }
+
+  async addTypedProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    propertyKey: string,
+    propertyType: PropertyType,
+    propertyValue: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ) {
+    return this.submitTransaction(
+      creator,
+      "add_typed_property",
+      [tokenType || this.tokenType, PropertyTypeMap[propertyType]],
+      [HexString.ensure(token).hex(), propertyKey, propertyValue],
+      extraArgs,
+    );
+  }
+
+  async updateTypedProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    propertyKey: string,
+    propertyType: PropertyType,
+    propertyValue: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ) {
+    return this.submitTransaction(
+      creator,
+      "update_typed_property",
+      [tokenType || this.tokenType, PropertyTypeMap[propertyType]],
+      [HexString.ensure(token).hex(), propertyKey, propertyValue],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Transfer a token ownership.
+   * We can transfer a token only when the token is not frozen (i.e. owner transfer is not disabled such as for soul bound tokens)
+   * @param owner The account of the current token owner
+   * @param token Token address
+   * @param recipient Recipient address
+   * @returns The hash of the transaction submitted to the API
+   */
+  async transferTokenOwnership(
+    owner: AptosAccount,
+    token: MaybeHexString,
+    recipient: MaybeHexString,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    const builder = new TransactionBuilderRemoteABI(this.provider.aptosClient, {
+      sender: owner.address(),
+      ...extraArgs,
+    });
+    const rawTxn = await builder.build(
```

Comment:
Let's follow up with another client for Objects specifically.  I suspect this will be useful for other purposes outside of NFTs